src/components/Lights/Lights.component.tsx

```diff
@@ -0,0 +1,109 @@
+import { ContactShadows } from '@react-three/drei';
+import { useThree } from '@react-three/fiber';
+import React, { FC, useEffect, useState } from 'react';
+import { LightingProps } from 'src/types';
+import { Object3D, Vector3 } from 'three';
+
+export const LIGHT_CONFIG = Object.freeze({
+  fillLightAngle: Math.PI / 3,
+  backLightAngle: Math.PI / 8,
+  keyLightAngle: Math.PI / 2,
+  keyLightPosition: new Vector3(0.5, 1.55, 0.5),
+  liftLightPosition: new Vector3(0.25, 1.7, 2.0),
+  defaultProps: {
+    keyLightIntensity: 1.2,
+    keyLightColor: '#e8e3df',
+    fillLightIntensity: 0.7,
+    fillLightColor: '#99ccff',
+    fillLightPosition: new Vector3(-0.5, 1.6, -0.5),
+    backLightIntensity: 2.3,
+    backLightColor: '#fff0d6',
+    backLightPosition: new Vector3(0.5, 1.6, -1.0),
+    lightTarget: new Vector3(0.0, 1.7, 0.0)
+  } as Required<LightingProps>
+});
+
+Object.fromEntries = (arr: Array<any>) => Object.assign({}, ...arr.map(([k, v]) => ({ [k]: v })));
+
+const definedProps = (obj: Object) =>
+  Object.fromEntries(
+    // eslint-disable-next-line
+    Object.entries(obj).filter(([k, v]) => v !== undefined)
+  );
```

Comment:
Why is this necessary?

Reply:
We use this polyfill for prop checking in the Lights component.

Reply:
Let's do it in a way that doesn't override the default Javascript method `Object.fromEntries` 🙏